apps/system/getfile_tool/getfile_main.c

```diff
@@ -0,0 +1,159 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/// @file getfile_main.c
+#include <tinyara/config.h>
+#include <tinyara/arch.h>
+#include <arch/irq.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+#include <fcntl.h>
+/****************************************************************************
+ * Definitions
+ ****************************************************************************/
+#define HANDSHAKE_STRING                "GETFILE"
+#define HANDSHAKE_STR_LEN_MAX			(7)
+#define FILEPATH_STR_LEN_MAX			(30)
+/****************************************************************************
+ * Functions
+ ****************************************************************************/
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int getfile_main(int argc, char *argv[])
+#endif
+{
+	int i;
+	FILE *fp;
+	int f_length = 0;
+	int ch;
+	uint8_t *ptr;
+	char *fbuf;
+	uint32_t size;
+	int psize = 0;
+	char *target_str = HANDSHAKE_STRING;
+	char host_buf[HANDSHAKE_STR_LEN_MAX] = "";
+	char file_path[FILEPATH_STR_LEN_MAX] = "";
+	irqstate_t flags;
+#if !defined(CONFIG_ARCH_LOWPUTC)
+	/* If lowlevel serial is not available, getfile is not possible */
```

Comment:
Actaully I don't understand why this feature is mandatory for getfile.
I think you might get this from RAMDUMP. RAMDUMP is working on crash status so that it can use only low level API like this.
But getfile is working on normal status. We don't need to use low level API. Additionally here is app context. up_xxx APIs are not for app context.

Reply:
Firstly,  some project may be has not  USB hardware so cannot use USB related functions to pull file. If we want pull file from target, We can only use UART.  After all, most of the hardware supports UART.  Secondly, If we use UART, we must use UART driver funtions, such as up_xxx APIs to pull character one by one. As you said, up_xxx is low level API, getfile is app function. But I am not sure whether there is any app level functions to realize serial port up_xxx

Reply:
1. I don't mean let's use USB instead of UART. There is no issue on this topic. (I don't know why you mention this actually)
2. I think you can use "/dev/console" with open, read, write APIs. Please refer tash_main.c
3. You set this as ASYNC and no configured for thread priority (It means this thread will have 100 
 priority). Is this working with TASH? Does not TASH take data from UART instead of getfile?